2018/opmodes/BachJewelAutonomous.java

```diff
@@ -0,0 +1,330 @@
+                    }
+                });
+            }
+        }
+        moveToSimplePark.addSegment(DeadReckonPath.SegmentType.STRAIGHT, 42, MOVE_MULTIPLIER * MOVE_SPEED);
+        moveToPark.addSegment(DeadReckonPath.SegmentType.STRAIGHT, 42, MOVE_MULTIPLIER * MOVE_SPEED);
+    }
+
+    public void handleGamepadSelection(GamepadTask.GamepadEvent event) {
+        switch (event.kind) {
+            case BUTTON_X_DOWN:
+                alliance = Alliance.BLUE;
+                persistentTelemetryTask.addData("ALLIANCE: ", "" + alliance);
+                break;
+            case BUTTON_B_DOWN:
+                alliance = Alliance.RED;
+                persistentTelemetryTask.addData("ALLIANCE: ", "" + alliance);
+                break;
+            case BUTTON_A_DOWN:
+                colorThiefTask.setPollingMode(ColorThiefTask.PollingMode.ON);
+                break;
+            case BUTTON_Y_DOWN:
+                colorThiefTask.setPollingMode(ColorThiefTask.PollingMode.OFF);
+                break;
+            case LEFT_BUMPER_DOWN:
+                startingPosition = StartingPosition.R1;
+                persistentTelemetryTask.addData("STARTING POSITION: ", "" + startingPosition);
+                break;
+            case LEFT_TRIGGER_DOWN:
+                startingPosition = StartingPosition.R2;
+                persistentTelemetryTask.addData("STARTING POSITION: ", "" + startingPosition);
+                break;
+            case RIGHT_BUMPER_DOWN:
+                startingPosition = StartingPosition.B1;
+                persistentTelemetryTask.addData("STARTING POSITION: ", "" + startingPosition);
+                break;
+            case RIGHT_TRIGGER_DOWN:
+                startingPosition = startingPosition.B2;
+                persistentTelemetryTask.addData("STARTING POSITION: ", "" + startingPosition);
+                break;
+        }
+    }
+
+    protected void red1Init() {
+        MOVE_MULTIPLIER = -1;
+        TURN_MULTIPLIER = 1;
+    }
+
+    protected void red2Init() {
+        MOVE_MULTIPLIER = -1;
+        TURN_MULTIPLIER = 1;
+    }
+
+    protected void blue1Init() {
+        MOVE_MULTIPLIER = 1;
+        TURN_MULTIPLIER = -1;
+    }
+
+    protected void blue2Init() {
+        MOVE_MULTIPLIER = 1;
+        TURN_MULTIPLIER = -1;
+    }
+
+    protected void detectParticle() {
+        colorThiefTask = new ColorThiefTask(this) {
+            @Override
+            public void handleEvent(RobotEvent e) {
+                ColorThiefTask.ColorThiefEvent event = (ColorThiefEvent) e;
+                // TODO: add telemetry
+                if (alliance == Alliance.RED) {
+                    switch (event.kind) {
+                        case RED:
+                            persistentTelemetryTask.addData("DETECTED COLOR", "RED");
+                            RobotLog.i("104 Detected red");
+                            pushJewel.addSegment(DeadReckonPath.SegmentType.STRAIGHT, 10, MOVE_MULTIPLIER * MOVE_SPEED);
```

Comment:
Once you turn on polling, this is going to cause a problem because the event handler will be called every two seconds which means you will add another segment every two seconds, which means you will end up with a big list of identical segments.  You need to call the stop() method of DeadReckonPath() before calling addSegment each time you enter handleEvent.   stop() is somewhat misnamed.  What it does is clear the path so that it has no segments.  We will fix the naming later.